### ISSUE TITLE
[BUG] Playlist Subscription will stop working if blocked video is encountered

### ISSUE THREAD

**Issue by FranciscoTrigo** [Tzahi12345/YoutubeDL-Material#928]
**Describe the bug**
The playlist subscription feature will stop working if it encounters a blocked video (copyright blocked in my country, in my case) and will not try to download any more videos after it.

**To Reproduce**
This is how it happened to me:
1. Watch a video that I liked and add it to my playlist that I download from
2. That video was copyright blocked in my country after I watched it but before it was downloaded
3. No more videos are downloaded now

I dont know if this would also happen with deleted videos or videos that are made private by the uploader

**Expected behavior**
Skip all videos that cant be downloaded for any reason and keep downloading every other video that it can in the playlist. 

**Environment**
Running on Unraid

- YoutubeDL-Material version 4.3
Installation type: docker
Docker tag: latest
Commit hash: fc3c179
Build date: 2022-07-05 

**Logs**
```
ERROR: [youtube] GlC9ongtH-4: Video unavailable. This video contains content from NBC Universal, who has blocked it in your country on copyright grounds
2023-05-25T05:49:51.280Z ERROR: Subscription check failed!
2023-05-25T05:55:30.080Z ERROR: WARNING: [youtube:tab] YouTube said: INFO - 1 unavailable video is hidden
ERROR: [youtube] GlC9ongtH-4: Video unavailable. This video contains content from NBC Universal, who has blocked it in your country on copyright grounds
2023-05-25T05:55:30.081Z ERROR: Subscription check failed!
```
**My workaround to fix issue**
I noticed the logs talking about a blocked video, so I went on my phone and connected to a VPN in Panama and compared available videos on the playlist until I found the culprit and removed it from the playlist.
After that, my videos kept downloading as they should

**Comment by Tzahi12345** [comment]
Can you provide the URL so I can test this on my end?

**Comment by Tzahi12345** [comment]
Figured out what's going on, will be fixed when #935 merges. Basically there exists logic to deal with missing videos, but that only applies to subscriptions. Worse, the logic we had made it so ignoring errors wouldn't fix the issue but make the download fail outright. Both those should be resolved now, the logic is now shared between subscriptions and one-off downloads.

**Comment by FranciscoTrigo** [comment]
Thank you, I didnt have the link to that video anymore, and I was trying to find another blocked video to give here 
Glad it got sorted out.

**Comment by ekadin-mtc** [comment]
I just ran into this issue on 4.3.2 (latest)
The URL for the playlist is https://www.youtube.com/playlist?list=PLRdw3IjKY2gkdyoTNmD4PmFoiuJ0QsEtU

`2023-10-30T03:56:28.566Z ERROR: WARNING: [youtube:tab] YouTube said: INFO - 1 unavailable video is hidden
WARNING: [youtube:tab] Incomplete data received. Retrying (1/3)...
WARNING: [youtube:tab] Incomplete data received. Retrying (2/3)...
WARNING: [youtube:tab] Incomplete data received. Retrying (3/3)...
WARNING: [youtube:tab] Incomplete data received. Giving up after 3 retries
ERROR: [youtube] EaCrF6mqV8g: Video unavailable. This video has been removed by the uploader
2023-10-30T03:56:28.567Z ERROR: Subscription check failed!`
